plugins/linux-v4l2/v4l2-codec.c

```diff
@@ -44,7 +56,10 @@ int v4l2_init_mjpeg(struct v4l2_mjpeg_decoder *decoder)
 		return -1;
 	}
 
-	decoder->context->flags2 |= AV_CODEC_FLAG2_FAST;
+	if (pixfmt == V4L2_PIX_FMT_MJPEG) {
+		decoder->context->flags2 |= AV_CODEC_FLAG2_FAST;
+		decoder->context->pix_fmt = AV_PIX_FMT_YUVJ422P;
```

Comment:
pixel format needs to be inferred from the decoder, this looks like it reintroduces breakages for other yuv formats.